subprojects/configuration-cache/src/integTest/groovy/org/gradle/configurationcache/ConfigurationCachePublishingIntegrationTest.groovy

```diff
@@ -65,9 +73,119 @@ class ConfigurationCachePublishingIntegrationTest extends AbstractConfigurationC
         """
     }
 
+    def "can publish maven publication metadata to remote repository"() {
+        def username = "someuser"
+        def password = "somepassword"
+        with(server) {
+            requireAuthentication(username, password)
+            // or else insecure protocol enforcement is skipped
+            useHostname()
+            start()
+        }
+        def remoteRepo = new MavenHttpRepository(server, mavenRepo)
+        def prepareRepository = {
+            def rootModule = remoteRepo.module("group", "root")
+            def credentials = new DefaultPasswordCredentials(username, password)
+            rootModule.pom.expectPublish(true, credentials)
+            rootModule.moduleMetadata.expectPublish(true, credentials)
+            rootModule.rootMetaData.expectGetMissing(credentials)
+            rootModule.rootMetaData.expectPublish(true, credentials)
+        }
+
+        def repositoryName = "CustomHttp"
+        settingsFile.setText "rootProject.name = 'root'"
+        buildFile << buildFileConfiguration("""
+            repositories {
+                maven {
+                    name "${repositoryName}"
+                    url "${remoteRepo.uri}"
+                    allowInsecureProtocol true
+                    credentials {
+                        username '${username}'
+                        password '${password}'
+                    }
+                }
+            }
+        """)
+        def configurationCache = newConfigurationCacheFixture()
+        def metadataFile = file('build/publications/maven/module.json')
+        def tasks = [
+            "generateMetadataFileForMavenPublication",
+            "generatePomFileForMavenPublication",
+            "publishMavenPublicationTo${repositoryName}Repository",
+            "publishAllPublicationsTo${repositoryName}Repository"
+        ]
+
+        expect:
+        !GUtil.isSecureUrl(server.uri)
+
+        when:
+        prepareRepository()
+        configurationCacheRun(*tasks)
+        server.resetExpectations()
+
+        then:
+        configurationCache.assertStateStored()
+        metadataFile.exists()
+
+        when:
+        def storeTimeRepo = mavenRepoFiles()
+        def storeTimeMetadata = metadataFile.text
+        metadataFile.delete()
+        deleteDirectory(mavenRepo.rootDir)
+
+        prepareRepository()
+        configurationCacheRun(*tasks)
+        server.resetExpectations()
+
+        then:
+        configurationCache.assertStateLoaded()
+        def loadTimeRepo = mavenRepoFiles()
+        storeTimeRepo == loadTimeRepo
+        def loadTimeMetadata = metadataFile.text
+        storeTimeMetadata == loadTimeMetadata
+    }
+
     def "can publish maven publication metadata to local repository"() {
```

Comment:
This is mostly existing code.

Not happy with all the duplication between the existing and new test case. Spock makes it challenging to share some assertion code though. Happy to hear suggestions.